src/resolve/resolved-dns-cache.c

```diff
@@ -54,6 +54,7 @@ struct DnsCacheItem {
         LIST_FIELDS(DnsCacheItem, by_key);
 
         bool shared_owner;
+        usec_t ttl_expiry;        /* The key is for storing the time when the TTL set to expire. */
```

Comment:
given this is closely related to "until", I think it should be defined close to it

Reply:
renamed to until_valid